Dont error on returned constant, fix for levit_128

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113389
* __->__ #112544
* #112543

Previously, levit_128 would fail on inference because we would return a view of a constant, which messed up our assertions of outputs being in the cuda graph pool.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

eellison committed:
Dont error on returned constant, fix for levit_128

[ghstack-poisoned]

eellison committed:
Update on "Dont error on returned constant, fix for levit_128"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

eellison committed:
Update on "Dont error on returned constant, fix for levit_128"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

ezyang commented on torch/_inductor/cudagraph_trees.py:
```diff
@@ -582,14 +585,18 @@ def run(self, new_inputs):
+                    and t.untyped_storage().data_ptr() not in existing_path_data_ptrs
+                ):
+                    non_cudagraph_inps.add(t.untyped_storage().data_ptr())
+            return non_cudagraph_inps
```
poor man's scoping?

eellison replied:
previously, i had this inlined and at the end `del t` but `t` is not always set... better suggestion ?

ezyang reviewed (approved):
I don't really understand but this seems plausible

eellison commented:
@pytorchbot merge

eellison commented:
@pytorchbot merge

eellison committed:
Update on "Dont error on returned constant, fix for levit_128"


Previously, levit_128 would fail on inference because we would return a view of a constant, which messed up our assertions of outputs being in the cuda graph pool.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

eellison commented:
@pytorchbot merge
